### PR TITLE
Support/random word special char guard

### LINKED ISSUE #17: Add additional guard to remove special characters from randomWord function
Ran into this random failure today:

https://travis-ci.com/github/brandongregoryscott/AndcultureCode.Cli/jobs/426634572

```
 ● Github › addTopicToRepository › given a topic, owner, and repoName › it calls the topic api and returns the updated topics
    SyntaxError: Invalid regular expression: /repos/Tasty/(customarily/topics/: Unterminated group
        at new RegExp (<anonymous>)
      28 |      */
      29 |     const getRepoTopicsRoute = (owner: string, repoName: string) =>
    > 30 |         new RegExp(
         |         ^
      31 |             [
      32 |                 Github.apiRepositoriesRouteParam,
      33 |                 owner,
      at getRepoTopicsRoute (src/modules/github.test.ts:30:9)
      at src/modules/github.test.ts:171:26
      at step (src/modules/github.test.ts:33:23)
      at Object.next (src/modules/github.test.ts:14:53)
      at src/modules/github.test.ts:8:71
      at Object.<anonymous>.__awaiter (src/modules/github.test.ts:4:12)
      at Object.<anonymous> (src/modules/github.test.ts:156:75)
```

Looks like the `randomWord` function returned parentheses, which caused an unterminated regular expression. I would not expect this function to return special characters (anything that's not a letter or number), just like it shouldn't return spaces.

Should be simple as a `.replace(/[^A-Za-z0-9]/gi,"")` - be sure to add applicable tests.

### PR DESCRIPTION
**Pull request by brandongregoryscott**
Fixes #17 Add additional guard to remove special characters from randomWord function 

- Updated the `randomWord` function to remove non-alphanumeric characters.
- Added a `randomWords` function to return an array of strings, as that has been a common use-case. 
- Added a `testLoop` spec for running a function body multiple times, to hopefully catch some of these edge cases. 
- Finally, flipped `removeComments` to `false` in the tsonfig & removed the private wrapper functions in `TestUtils` so that JSDoc comments are actually retained.

- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [-] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: src/utilities/shared-specs.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * Utility function for running a test body a certain number of times. Useful for ensuring specific
+ * behavior on implementations that return randomized data and edge cases will not immediately be
+ * exposed.
+ *
+ * @param {string} name Name of the test
+ * @param {Function} fn Function body to run (can be asynchronous)
+ * @param {number} [times=1000] Number of times to run the test function.
+ */
+const testLoop = (name: string, fn: Function, times: number = 1000) => {
```
Comment (mrjones2014) [inline src/utilities/shared-specs.ts:14]: maybe consider lowering the default value for `times`; I feel like for most things, around 100 should be enough, and it would make the tests run faster.

**Review by mrjones2014 — APPROVED** [review]

File: src/utilities/shared-specs.ts (hunk 1 above)
Comment (brandongregoryscott) [reply src/utilities/shared-specs.ts:14]: Fair enough 😛 I'll lower it and swap some implementations to make sure it's still catching these patch cases we have

**Review by mrjones2014 — APPROVED** [review]
